### ISSUE TITLE
loadbridge uses property file before open()

### ISSUE THREAD

**Issue by svtesan** [OpenMAMA/OpenMAMA#37]
During creation of the default queue, properties subsystem is accessed and this always prints an error/warning if we use a custom properties file.

**Comment by fquinner** [comment]
What middleware bridge are you using? That's typically where the default queue is created.

Note the order of properties loading changed in 2.4.0 so there is a good chance this might already be resolved by that change.

If the issue is still at large, can you provide recreation details please?

**Comment by svtesan** [comment]
Sorry for being sparse on details. I think this should for any bridge - but the bridge in question is qpid. The issue is in mama_loadBridgeWithPathInternal(). It calls mamaBridgeImpl::bridgeOpen once it loads the bridge. And in most bridge open()s, you need to create a default queue using mamaQueue_create() and this tries to access the properties - since this happens during bridge load (before mama_openWithProperties() can be called) there is a warning printed in mamaInternal_loadProperties
Code flow:
mama_loadBridge -> mama_loadBridgeWithPathInternal -> mamaBridgeImpl::bridgeOpen -> mamaQueue_create -> checks a property ama_getProperty(MAMAQUEUE_PROPERTY_OBJECT_LOCK_TRACKING) -> gProperties is NULL at this stage so calls mamaInternal_loadProperties(NULL, NULL) - > complains that it failed to open properties file

The control flow is intact even in 2.4.0, so it is an (minor) issue still.  Internally, I patched mama.c to log at ERROR only if the path was extracted from env var - altleast to get rid of an ERROR level log. We monitor all logs for ERRORs so this spurious log is really an issue for us

**Comment by svtesan** [comment]
Index: mama.c
===================================================================
--- mama.c	(revision 19648)
+++ mama.c	(revision 19649)
@@ -282,6 +282,9 @@
             WOMBAT_PATH_ENV);
     }
 
+	//do not complain if just trying to check if properties are found by  passing NULL, NULL
+	int complain = path && filename;
+
     if( !filename )
     {
        filename = PROPERTY_FILE;
@@ -297,7 +300,11 @@
 
     if( fileProperties == 0 )
     {
-			mama_log (MAMA_LOG_LEVEL_ERROR, "Failed to open properties file.\n");
+		if (complain)
+			mama_log(MAMA_LOG_LEVEL_ERROR, "Failed to open properties file.\n");
+		else
+			mama_log(MAMA_LOG_LEVEL_NORMAL, "No default mama.properties found");
+
         return;
     }

**Comment by fquinner** [comment]
This is a tricky enough one. It's not just this particular issue it's the fact that unless you're using the default property file name, the bridge will be unable to pick up any additional configuration parameters.

As a workaround, there is the option of calling
```
mama_setPropertiesFromFile (const char *path, const char *filename);
```
Before calling mama_load* function. It'll report an error but it will functionally work. I think something like that will be required anyway because the loadBridge functions are where the bridges get an opportunity to set themselves up so the bridges will need properties access set up before that. They have it now where they didn't have it before as of 2.4.x, but not for custom configuration files like this.

I'm thinking the best solution may be something like:
```
mama_setDefaultPropertiesFile (const char* filename);
```
Which can be called by the application. A `NULL` value means no default properties will be loaded and it will instead rely on functions like `mama_setProperty` (perhaps to map from an alternative configuration file format).

Another option is you have a function like `mama_disableDefaultProperties` which you could use in conjunction with the existing `mama_setPropertiesFromFile` to have the same affect.

**Comment by fquinner** [comment]
Working on a solution here that can work just with mama_setPropertiesFromFile. It will only report an error if:

* A specific properties file was requested
* Default properties are being loaded *and* no existing properties are loaded

**Comment by fquinner** [comment]
Implemented in #208
